packages/babel-types/src/validators/isReferenced.js

```diff
@@ -66,6 +67,13 @@ export default function isReferenced(
       if (parent.value === node) {
         return !grandparent || grandparent.type !== "ObjectPattern";
       }
+      if (parent.params) {
+        for (const param of parent.params) {
+          if (param === node) {
+            return false;
+          }
+        }
```

Comment:
What about using `parent.params.indexOf(node) !== -1`?

Reply:
Ah, yeah, that's much better. Updated.